src/Command.php

```diff
@@ -60,27 +60,23 @@ public function getHandlerList()
         return $this->handlerList;
     }
 
-    /**
-     * For overriding auth schemes on a per endpoint basis when using
-     * EndpointV2 provider. Intended for internal use only.
-     *
-     * @param array $authSchemes
-     *
-     * @internal
-     */
-    public function setAuthSchemes(array $authSchemes)
-    {
-        $this->authSchemes = $authSchemes;
-    }
-
     /**
      * Get auth schemes added to command as required
      * for endpoint resolution
      *
      * @returns array
+     *
+     * @deprecated In favor of using the @context property bag.
+     *             Auth schemes are now accessible via the `signature_version` key
+     *             in a Command's context, if applicable.
      */
     public function getAuthSchemes()
     {
+        trigger_error(__METHOD__ . ' is deprecated.  Auth schemes '
+        . 'resolved using the service `auth` trait or via endpoint resolution '
+        . 'can now be found in the command `@context` property.`'
+        , E_USER_DEPRECATED);
+
         return $this->authSchemes ?: [];
```

Comment:
This is an attempt to walk back a poor design choice made during the initial EP 2.0 implementation.  The `authSchemes` property was only set (using the preceding method) _after_ endpoint resolution if the resolved endpoint contained auth schemes. If an auth scheme was resolved from the model, whether it be the operation or at the service metadata level, this would return nothing.

Reply:
Endpoint resolution based auth should continue setting this content and just not using it, a method overload could be added to suppress the deprecation notice for that call. The method should continue functioning as it did for its case while deprecated.

Reply:
Added an error suppression operator to the call, but definitely open to overriding `__call` if you think we should go that route